mlrun/datastore/redis.py

```diff
@@ -41,16 +41,12 @@ def __init__(
         super().__init__(parent, name, schema, endpoint)
         self.headers = None
 
-        self.endpoint = self.endpoint or mlrun.mlconf.redis_url
-
         if self.endpoint.startswith("rediss://"):
             self.endpoint = self.endpoint[len("rediss://") :]
             self.secure = True
         elif self.endpoint.startswith("redis://"):
             self.endpoint = self.endpoint[len("redis://") :]
             self.secure = False
-        else:
-            raise NotImplementedError(f"invalid endpoint: {endpoint}")
```

Comment:
Isn't there any validation that we can do here? For example, at this point the `endpoint` may be empty, which shouldn't be valid. Also, I don't know if we always need a port specified or if there's a default port?

Reply:
added a check for empty endpoint.